test/osutils.jl

```diff
@@ -44,6 +44,9 @@ end
     @test (@static if false 1 elseif false 2 else 3 end) === 3
     @test (@static if false 1 elseif false 2 elseif true && false 3 else 4 end) === 4
     @test (@static if false 1 elseif false 2 elseif true && false 3 end) === nothing
+    @test_throws ArgumentError(
+        "invalid @static macro"
+    ) var"@static"(LineNumberNode(@__LINE__), @__MODULE__, 1)
```

Comment:
I think this would be more typical (for syntax and macrocall-test style)?
```suggestion
    @test_throws(LoadError(@__FILE__, @__LINE__, ArgumentError("invalid @static macro")),
                 @macroexpand @static(1))
```

Reply:
Sure. I'm not sure either version looks much more elegant than the other, but happy to change. Maybe there should be a dedicated `@test_macro_throws` or somesuch?

Reply:
True, both are long lines. Another option is:
```suggestion
    expected = LoadError(@__FILE__, @__LINE__, ArgumentError("invalid @static macro"))
    @test_throws expected @macroexpand @static(1))
```

Reply:
Ref https://github.com/JuliaLang/julia/pull/34854